### PR TITLE
LPS-94125 Criteria editor usability improvements

### PR DESCRIPTION
**Pull request by dgarciasarai**
Fix: LPS-94123, LPS-94125, LPS-941267 and LPS-94536

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Review with @andresfulla ;)

**Comment by dgarciasarai** [comment]
ci:test:sf

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 1]
```diff
@@ -115,17 +124,59 @@ class ContributorBuilder extends React.Component {
 			conjunctionName: CONJUNCTIONS.AND,
 			contributors,
 			editingId: propertyKey
+			membersCount: this.props.membersCount,
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: props.membersCount

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 2]
```diff
@@ -155,6 +208,29 @@ class ContributorBuilder extends React.Component {
...
+					destroyOnHide: true
+				},
+				id: 'segment-members-dialog',
+				title: sub(Liferay.Language.get('x-members'), [this.props.values.name]),
+				uri: url
+			}
+		);
+	}
+
+	_handleSelectorChange = event => {
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js]: Are we using this?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js [hunk 3]
```diff
@@ -161,9 +108,9 @@ class SegmentEdit extends Component {
 	);
 
 	_renderContributors = () => {
-		const {contributors, propertyGroups} = this.props;
+		const {contributors, formId, previewMembersURL, propertyGroups, requestMembersCountURL, values} = this.props;
 
-		const {editing} = this.state;
+		const {editing, membersCount} = this.state;
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js]: looks to me that members count lives in the state on the component a level down 👀

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaBuilder.es.js [hunk 4]
```diff
@@ -268,18 +265,26 @@ class CriteriaBuilder extends Component {
 		} = this.props;
 
 		const criteriaBuilderClassNames = getCN(
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaBuilder.es.js]: inline className will suffice

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/__snapshots__/CriteriaRow.es.js.snap [hunk 5]
```diff
@@ -24,7 +24,7 @@ exports[`CriteriaRow should render 1`] = `
       <span
         class="criterion-string"
       >
-        x-with-property-x
+        <b />
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/test/js/components/criteria_builder/__snapshots__/CriteriaRow.es.js.snap]: I don't think this is valid markup.

**Comment by dgarciasarai** [comment]
ci:test:sf

**Pull request closed without merging**